### PR TITLE
feat: move related tags concatenation logic EX-5410 

### PR DESCRIPTION
**Pull request by javieri-empathy**
- Standardises `query` and  `setQuery` in all modules.
- Related tag concatenation is now done in the x-components.
- BREAKING CHANGE: Removes `previous` and `selected` properties from `RelatedTag` model.
- BREAKING CHANGE Remove related tags query mapper from the `x-adapter` package.

### CODE REVIEW

File: packages/search-adapter/src/empathy/__tests__/request-mappers/empathy-queryable-request-mapper.spec.ts [hunk 1]
```diff
@@ -1,19 +1,13 @@
-import { RelatedTag } from '@empathyco/x-types';
 import { Container } from 'inversify';
```
Comment (javieri-empathy) [inline packages/search-adapter/src/empathy/__tests__/request-mappers/empathy-queryable-request-mapper.spec.ts:1]: As adapter files have been auto-formatted by prettier they are going to be a little harder to review, but the intention was to only remove the related tags  mapping from the request mappers.

File: packages/x-components/src/__stubs__/related-tags-stubs.factory.ts [hunk 2]
```diff
@@ -15,15 +15,6 @@ export function getRelatedTagsStub(amount = 12): RelatedTag[] {
   );
 }
 
-/**
- * Function to create selected related tags stub.
- *
- * @returns Array of selected related tags stub.
- */
-export function getSelectedRelatedTagsStub(): RelatedTag[] {
```
Comment (javieri-empathy) [inline packages/x-components/src/__stubs__/related-tags-stubs.factory.ts:23]: We don't need this function anymore, there is no difference in the model of a selected related tag or an unselected one.

File: packages/x-components/src/x-modules/search/store/index.ts [hunk 3]
```diff
@@ -2,5 +2,6 @@ export * from './actions/fetch-and-save-search-response.action';
 export * from './actions/fetch-search-response.action';
```
Comment (javieri-empathy) [inline packages/x-components/src/x-modules/search/store/index.ts:2]: There is a big mess with x-module exports. There are actions or getters that are not exported, and some times the name even collides with the wires. I'm planning to fix this in another PR.

File: packages/x-components/src/x-modules/search/store/types.ts [hunk 4]
```diff
@@ -73,14 +74,16 @@ export interface SearchGetters {
   /** The adapter request object for retrieving the results, or null if there is not
    * valid data to create a request. */
   request: InternalSearchRequest | null;
+  /** The combination of the query and the selected related tags. */
+  query: string;
```
Comment (javieri-empathy) [inline packages/x-components/src/x-modules/search/store/types.ts:78]: I decided to leave this `query` and the `setQuery` mutation in every module so we can link it through the documentation.

Comment (diegopf) [reply packages/x-components/src/x-modules/search/store/types.ts:78]: I was finishing the review and I just saw this, you are so good that answer my questions before I asked them 🤣

**Review by diegopf — COMMENTED** [review]
LGTM, I'm just adding a comment to find out about the EX-2163 issue and if we can remove the TODO comments

File: packages/search-adapter/src/types/requests.types.ts [hunk 5]
```diff
@@ -68,15 +67,20 @@ export interface RequestOptions {
  * @public
  */
 export interface ExtraParamsRequest {
-  [key: string]: unknown
+  [key: string]: unknown;
 }
 
 /**
  * TODO https://searchbroker.atlassian.net/browse/EX-2163
```
[the remark this answers is not in the document]
Comment (javieri-empathy) [reply packages/search-adapter/src/types/requests.types.ts:74]: That issue was just "Document the whole adapter"

**Review by guillei10 — APPROVED** [review]

File: packages/x-components/src/x-modules/history-queries/store/types.ts [hunk 6]
```diff
@@ -57,7 +58,7 @@ export interface HistoryQueriesGetters {
  *
  * @public
  */
-export interface HistoryQueriesMutations {
+export interface HistoryQueriesMutations extends QueryMutations {
```
[the remark this answers is not in the document]
Comment (javieri-empathy) [reply packages/x-components/src/x-modules/history-queries/store/types.ts:61]: Just answering here too for the future haha, but yeah, my reason was that with API extractor we can't link to interface inherited properties, we have to link the base interface, and I don't like that in the documentation. As types do not affect runtime performance, I think we can afford some duplication just to provide better API documentantion. But I would love to hear your opinion on this.

**Review by diegopf — APPROVED** [review]

**Pull request merged**